net/tailscale/files/tailscale.init

```diff
@@ -1,47 +1,250 @@
 #!/bin/sh /etc/rc.common
 
-# Copyright 2020 Google LLC.
-# Copyright (C) 2021 CZ.NIC z.s.p.o. (https://www.nic.cz/)
-# SPDX-License-Identifier: Apache-2.0
+START=90
 
 USE_PROCD=1
-START=80
 
-start_service() {
-  local state_file
-  local port
-  local std_err std_out
+PROG=/usr/sbin/tailscale
+PROGD=/usr/sbin/tailscaled
+CONFIG_PATH=/var/lib/tailscale
 
-  config_load tailscale
-  config_get_bool std_out "settings" log_stdout 1
-  config_get_bool std_err "settings" log_stderr 1
-  config_get port "settings" port 41641
-  config_get state_file "settings" state_file /etc/tailscale/tailscaled.state
-  config_get fw_mode "settings" fw_mode nftables
+service_triggers() {
+  procd_add_reload_trigger "tailscale"
+  procd_add_interface_trigger "interface.*.up" wan /etc/init.d/tailscale reload
+}
 
-  /usr/sbin/tailscaled --cleanup
+section_enabled() {
+  config_get_bool enabled "$1" 'enabled' 0
+  [ $enabled -gt 0 ]
+}
+
+custom_instance() {
+  local cfg="$1"
+  local acceptRoutes hostname acceptDNS advertiseExitNode exitNode advertiseRoutes s2s subnetRoutes flags loginServer authkey std_out std_err
+  local ARGS=" up --reset"
+
+  config_get_bool acceptRoutes $cfg 'acceptRoutes'
+  config_get hostname $cfg 'hostname'
+  config_get_bool acceptDNS $cfg 'acceptDNS'
+  config_get_bool advertiseExitNode $cfg 'advertiseExitNode'
+  config_get exitNode $cfg 'exitNode'
+  config_get advertiseRoutes $cfg 'advertiseRoutes'
+  config_get_bool s2s $cfg 's2s'
+  config_get flags $cfg 'flags'
+  config_get loginServer $cfg 'loginServer'
+  config_get authkey $cfg 'authkey'
+  config_get_bool std_out $cfg 'log_stdout'
+  config_get_bool std_err $cfg 'log_stderr'
+
+  [ "$acceptRoutes" = "1" ] && ARGS="$ARGS --accept-routes=true"
+  [ -n "$hostname" ] && ARGS="$ARGS --hostname=$hostname"
+  [ "$acceptDNS" = "0" ] && ARGS="$ARGS --accept-dns=false"
+  [ "$advertiseExitNode" = "1" ] && ARGS="$ARGS --advertise-exit-node"
+  [ -n "$exitNode" ] && ARGS="$ARGS --exit-node=$exitNode --exit-node-allow-lan-access=true"
+  [ -n "$advertiseRoutes" ] && ARGS="$ARGS --advertise-routes=$(echo $advertiseRoutes | tr ' ' ',')"
+  [ "$s2s" = "1" ] && ARGS="$ARGS --snat-subnet-routes=false"
```

Comment:
Perhaps it would be clearer to name config variables same as CLI arguments:

```suggestion
  [ "$snat_subnet_routes" = "1" ] && ARGS="$ARGS --snat-subnet-routes=false"
```

Also the logic seems to be inverted here: when `1` set to `false`.